dev/create-release/release-build.sh

```diff
@@ -432,14 +432,14 @@ if [[ "$1" == "publish-snapshot" ]]; then
   echo "</server></servers></settings>" >> $tmp_settings
 
   if [[ $PUBLISH_SCALA_2_12 = 1 ]]; then
-    $MVN --settings $tmp_settings -DskipTests $SCALA_2_12_PROFILES $PUBLISH_PROFILES clean deploy
+    $MVN --settings $tmp_settings -DskipTests -DskipTests -Dmaven.javadoc.skip=true $SCALA_2_12_PROFILES $PUBLISH_PROFILES clean deploy
   fi
 
   if [[ $PUBLISH_SCALA_2_13 = 1 ]]; then
     if [[ $SPARK_VERSION < "4.0" ]]; then
       ./dev/change-scala-version.sh 2.13
     fi
-    $MVN --settings $tmp_settings -DskipTests $SCALA_2_13_PROFILES $PUBLISH_PROFILES clean deploy
+    $MVN --settings $tmp_settings -DskipTests -Dmaven.javadoc.skip=true -Dmaven.scaladoc.skip=true $SCALA_2_13_PROFILES $PUBLISH_PROFILES clean deploy
```

Comment:
I considered this option .. but I think we should include them ideally. e.g., if you include SNAPSHOT, and your IDE won't show the relevant method documentation anymore.

Reply:
Yes, you're correct. Even if it can be published successfully in the end, I think this is still a temporary solution. Since the final logs of the master's deploy task always stay at the doc build stage, I wonder if it would succeed by skipping this stage, also to provide some ideas for the next step of the fix.

Reply:
Could we try this solution? I don't have a better idea at the moment.

Reply:
Can we package first, and then run deploy?

Reply:
running `mvn` twice.

Reply:
only when `GITHUB_ACTIONS` variable is set.

Reply:
lIke

```
    $MVN --settings $tmp_settings -DskipTests $SCALA_2_12_PROFILES $PUBLISH_PROFILES clean package
    $MVN --settings $tmp_settings -DskipTests $SCALA_2_12_PROFILES $PUBLISH_PROFILES deploy
```
?

Let me test it locally first.

Reply:
👌

Reply:
The new code perform a two-step deployment (first package, then deploy) when `PACKAGE_BEFORE_DEPLOY` is true. However, after observing the compile log, I found that the second step's `scala:4.7.1:compile`, and `scala:4.7.1:doc-jar` are not skipped. I'm unsure if this change can really solve the problem...

Reply:
Maven doesn't behave like SBT, it won't directly deploy the already packaged jars even after being split into two commands.